Sources/EmbraceConfiguration/EmbraceConfigurable.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Copyright © 2024 Embrace Mobile, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// This protocol is used to add configuration to the runtime of the SDK
+/// It is used to configure the ongoing behavior of the SDK
+@objc public protocol EmbraceConfigurable {
+    var isSDKEnabled: Bool { get }
+
+    var isBackgroundSessionEnabled: Bool { get }
+
+    var isNetworkSpansForwardingEnabled: Bool { get }
+
+    var internalLogLimits: InternalLogLimits { get }
+
+    var networkPayloadCaptureRules: [NetworkPayloadCaptureRule] { get }
+
+    /// Tell the configurable implementation it should update if possible.
+    /// - Parameters:
+    ///     - completion: A completion block that takes two parameters (didChange, error). Completion block should pass `true` 
+    ///     if the configuration now has different values and `false` if not in the case of an error updating, the completion block should
+    ///     return `false` and an Error object describing the issue.
+    func update(completion: @escaping (Bool, Error?) -> Void)
```

Comment:
Wouldn't be cleaner something like `Result<Bool, Error>`. In that case, it’s easier to understand that:
- `.success` will be either `true` (was updated) or `false` (wasn't updated).
- `.failure` will always mean that something went wrong (and obviously, an update couldn't be done). 
As far as I can see, the remote and default config would work perfectly fine with this typification. Also, this approach prevents a peculiar case where you could have an error, yet still indicate that the update was done successfully (which seems contradictory)

Reply:
Yea I agree, the issue is that this protocol is exposed to objc and the `Result` enum is not available